### DISCUSSION TITLE
Enhance medusa.config.js to allow for Admin-UI config

### DISCUSSION THREAD

**Discussion by jaydubb12** [medusajs/medusa#5298] [Feature Requests]
# Feature Summary
Allow for a user to set / feature flag capabilities from the Admin-UI in a low code / configurable fashion to allow for an improved developer experience.

# Proposed Solution

## 1. Enhance the site build process and `medusa-config.js`
Utilize this approach to allow for core features to be easily feature flagged vs only the BETA features (i.e. Toggle on / off features such as Gift Cards, Discounts...)

## 2. Decompose admin into additional modules
An alternative solution approach is to break out these capabilities similar to the Stock, Inventory / Warehouse management features which can be implemented via
* Add additional module(s) to `package.json`
* Add reference to dependency within the `medusa-config.js` file

# Similar Discussions
* https://github.com/medusajs/medusa/discussions/4992
* https://github.com/medusajs/medusa/discussions/5093
* https://github.com/medusajs/medusa/discussions/5100
* https://github.com/medusajs/medusa/discussions/4116

# Benefits to platform / solution
* Improved developer experience / solution flexiibility
* Reduced confusion / complexity

# Current Solutions
## 1. Fork Repository
* Creates developer complexity
* Increases risk of code conflicts, and puts user in position similar to managing a monolith

## 2. Using `patch-package` to manage changes
* Reference Doc / Guidance - https://github.com/medusajs/medusa/discussions/5199
* Creates developer complexity
* Prevents effective tree shaking / library size reduction
* Increases risk of code conflicts, and puts user in position similar to managing a monolith

**Comment by pevey** [comment]
Another way to go about this, and the one that Magento uses, is to address this with roles.

For each role, the super admin defines what core functions users with that role have access to.  The super admin level role has access to anything, but your admin level role just under that could have certain things turned off, like settings, gift cards, whatever.  User level roles under that might have even more areas cut off.

**Comment by jaydubb12** [comment]
@pevey Agree that user roles is one way to do it, but I would respectfully contest that approach would not satisfy the feature enhancement / proposed benefits. 

### I believe the most impactful of the proposed benefits would be:
1. Reduced size of the library given only needed modules would be loaded

2. Improved ability to independently develop & release modules independently vs building / releasing the entire admin console

3. Improved ability for variations of the capabilities / functionalities via a plugin / add on construct that can be administered via the `medusa-config.js`
